packages/webapp/src/routes/credits-estimation/credits-tier.js

```diff
@@ -1,9 +1,41 @@
 import creditsEstimationConstants from '../../utils/credits-estimation-constants.js'
 import calculateCost from '../../credits/calculate.js'
 import Joi from 'joi'
+import { creditsValidationFailAction, creditsValidationSchema } from '../../utils/helpers.js'
 
-const errorMessage = { text: 'Enter the number of credits from the Metric up to 2 decimal places, like 23.75.' }
-const inputSchema = Joi.string().regex(/^\d*(\.\d{1,2})?$/).allow('')
+const defaultErrorMessage = { text: 'Enter at least one credit from the metric up to 2 decimal places, like 23.75' }
+const charLengthErrorMessage = { text: 'Number of credits must be 10 characters or fewer' }
+const inputSchema = Joi.string().max(10).regex(/^\d*(\.\d{1,2})?$/).allow('')
+
+const handlers = {
+  get: async (request, h) => {
+    const previousCostCalculation = request.yar.get(creditsEstimationConstants.redisKeys.ESTIMATOR_CREDITS_CALCULATION)
+    const inputValues = (previousCostCalculation)
+      ? Object.fromEntries(previousCostCalculation.tierCosts.map(({ tier, unitAmount, _ }) => [tier, unitAmount]))
+      : {}
+    return h.view(creditsEstimationConstants.views.ESTIMATOR_CREDITS_TIER, { inputValues })
+  },
+  post: async (request, h) => {
+    request.yar.set(creditsEstimationConstants.redisKeys.ESTIMATOR_CREDITS_CALCULATION, calculateCost(request.payload))
```

Comment:
just for my understanding, do you know if value is set before validation or after validation? after validation i guess!

Reply:
the value is set only if validation passes - if validation fails then an Error is thrown (see helpers line 801).